js/search/domain.js

```diff
@@ -22,7 +20,6 @@ const LINK_TABLE_ID = '4dba9091495e8f277893e0d400e9e092f9f6f551';
  */
 const Domain = new Lang.Class({
     Name: 'Domain',
-    Abstract: true,
```

Comment:
Have some semi formed thoughts here.

With the flatpak sandbox I think we could try to have a EKN_VERSION strongly coupled to a runtime version, so that all apps built with version say 3.5 of our runtime *have* to have content with EKN_VERSION 4. But particularly with global search and perhaps more generally, I can see us getting into a position where our content library needs to support multiple on disk formats of our offline content again.

However, a new version of our offline format would probably be coming from new requirements on our content, and I a little afraid of trying to predict the future here. Our current API surface on domains is fairly big
```
load_sync
get_domain_query_params
get_fixed_query[_finish]
resolve_xapian_result[_finish]
get_objects_by_query[_finish]
get_object_by_id[_finish]
check_for_updates
get_subscription_entries
```
If we have some new assumption-breaking content requirement that requires a new on disk format, how confident are we that API holds somewhat stable? And that domain continues to be the one stop shop for handle on disk format dependencies? Taking the full abstract plus subclass approach is gonna be a lot of boilerplate in C, which is totally fine that's what I signed up for, but I don't want us carry a lot of code we don't currently need unless we are reasonably confident in its longevity.

And of course we could keep around just the `get_domain_impl()` and a switch that errors if not version 3 if we want just that piece.

@magcius might have some thoughts here...